server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -1590,6 +1680,171 @@ void restoreShouldNotAppendMessagesToAnOtherUserMailbox() throws Exception {
+                    .body("statusCode", is(400))
+                    .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                    .body("message", is(notNullValue()));
+            }
+
+            @Test
+            void exportShouldReturnBadRequestWhenShareeIsInvalid() {
+                given()
+                    .queryParam("action", "export")
+                    .queryParam("sharee", "share@to#me@")
+                    .body(MATCH_ALL_QUERY)
+                .when()
+                    .post(USER.asString())
+                .then()
+                    .statusCode(HttpStatus.BAD_REQUEST_400)
+                    .body("statusCode", is(400))
+                    .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                    .body("message", is(notNullValue()))
+                    .body("details", is(notNullValue()));
+            }
+        }
+
+        @Nested
+        class TaskGeneratingTest {
+
+            @Test
+            void exportShouldReturnATaskCreated() {
+                given()
+                    .queryParam("action", "export")
+                    .queryParam("sharee", "sharee@james.org")
+                    .body(MATCH_ALL_QUERY)
+                .when()
+                    .post(USER.asString())
+                .then()
+                    .statusCode(HttpStatus.CREATED_201)
+                    .body("taskId", notNullValue());
+            }
+
+            @Test
+            void exportShouldProduceASuccessfulTaskWithInformation() {
+                vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+                vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
+
+                String taskId =
+                    with()
+                        .queryParam("action", "export")
+                        .queryParam("sharee", USER_2.asString())
+                        .body(MATCH_ALL_QUERY)
+                        .post(USER.asString())
+                    .jsonPath()
+                        .get("taskId");
+
+                given()
+                    .basePath(TasksRoutes.BASE)
+                .when()
+                    .get(taskId + "/await")
+                .then()
+                    .body("status", is("completed"))
+                    .body("taskId", is(taskId))
+                    .body("type", is(DeletedMessagesVaultExportTask.TYPE))
+                    .body("additionalInformation.userExportFrom", is(USER.asString()))
+                    .body("additionalInformation.sharee", is(USER_2.asString()))
+                    .body("additionalInformation.totalSharedMessages", is(2))
+                    .body("startedDate", is(notNullValue()))
+                    .body("submitDate", is(notNullValue()))
+                    .body("completedDate", is(notNullValue()));
+            }
+        }
+
+        @Nested
+        class SharingTest {
```

Comment:
you don't check you send the right email?

Reply:
checking emails is sent to the right address is over of the scope of webadmin test. Instead I had wrote a test to verify blobsharing mechanism share to to right address by mockito verifying: `exportShouldCallBlobSharingTargetToSharee`

Reply:
Here we abstract the side effect. We just ensure the side effect is triggered with the right parameter wich IMO is enough

Reply:
Do we plan to write an integration test for that?

Reply:
yeah it will be in the ticket of intergration test for exporting